File: src/main/java/com/seaweed/identifier/config/CustomSessionListener.java
package com.seaweed.identifier.config;

import com.seaweed.identifier.login.LoginService;
import com.seaweed.identifier.login.LoginVO;
import com.seaweed.identifier.user.vo.UserVO;
import jakarta.servlet.http.HttpSessionEvent;
import jakarta.servlet.http.HttpSessionListener;
import org.springframework.context.annotation.Configuration;

@Configuration
public class CustomSessionListener implements HttpSessionListener {
    @Override
    public void sessionCreated(HttpSessionEvent se) {
    }

    @Override
    public void sessionDestroyed(HttpSessionEvent se) {
        var session = se.getSession();
        if (session != null) {
            if(session.getAttribute("auth") != null){
                LoginVO user = (LoginVO)session.getAttribute("auth");
                LoginService.loginSessionMap.remove(user.getSeq(),session);
            }
        }
    }
}

File: src/main/kotlin/com/seaweed/identifier/config/CustomSessionListener.kt
package com.seaweed.identifier.config

import com.seaweed.identifier.auth.Authservice
import com.seaweed.identifier.user.User
import io.github.oshai.kotlinlogging.KotlinLogging
import jakarta.servlet.http.HttpSession
import jakarta.servlet.http.HttpSessionEvent
import jakarta.servlet.http.HttpSessionListener
import org.springframework.context.annotation.Configuration


private val logger = KotlinLogging.logger {  }
@Configuration
class CustomSessionListener : HttpSessionListener{
    override fun sessionCreated(se: HttpSessionEvent?) {
    }

    override fun sessionDestroyed(se: HttpSessionEvent?) {
        var session = se?.session;
        if (session != null) {
            if(session.getAttribute("auth") != null){
                var user = session.getAttribute("auth") as User
                user.seq?.let { Authservice.loginSessionMap.remove(it,session) }
            }
        }
    }
}